src/ctrv_model.cpp

```diff
@@ -0,0 +1,61 @@
+#include <cmath>
+#include <units.h>
+#include "cooperative_perception/ctrv_model.hpp"
+#include "cooperative_perception/utils.hpp"
+#include "cooperative_perception/units.hpp"
+
+namespace cooperative_perception
+{
+
+auto nextState(const CtrvState& state, units::time::second_t time_step) -> CtrvState
+{
+  using namespace units::literals;
+
+  units::length::meter_t delta_pos_x;
+  units::length::meter_t delta_pos_y;
+
+  if (utils::almostEqual(units::unit_cast<double>(state.yaw_rate), 0.0))
+  {
+    delta_pos_x = state.velocity * units::math::cos(state.yaw) * time_step;
+    delta_pos_y = state.velocity * units::math::sin(state.yaw) * time_step;
+  }
+  else
+  {
+    delta_pos_x = state.velocity / state.yaw_rate *
+                  (units::math::sin(state.yaw + state.yaw_rate * time_step) - units::math::sin(state.yaw)) * 1_rad;
+    delta_pos_y = state.velocity / state.yaw_rate *
+                  (-units::math::cos(state.yaw + state.yaw_rate * time_step) + units::math::cos(state.yaw)) * 1_rad;
```

Comment:
Is the 1_rad multiple just to set the unit?

Reply:
Yes. the `units::math::sin` and `units::math::cos` functions strip the units, which leaves a "dangling" radian.